src/client/analyticsFetchClient.ts

```diff
@@ -22,25 +22,29 @@ export class AnalyticsFetchClient implements AnalyticsRequestClient {
             ...(preprocessRequest ? await preprocessRequest(defaultOptions, 'analyticsFetch') : {}),
         };
 
-        const response = await fetch(url, fetchData);
-        if (response.ok) {
-            const visit = (await response.json()) as AnyEventResponse;
+        try {
+            const response = await fetch(url, fetchData);
```

Comment:
I would put the try catch only around the `fetch` specifically.
```suggestion
        try {
            const response = await fetch(url, fetchData);
        } catch (error) {
            console.error('An error has occured when sending the event.', error);
        }
```

Because they way you have it now, you'll catch the other thrown error on line 42 and you'll potentially display a weird message.
You'd catch this:

```
throw new Error(
                    `An error has occurred when sending the "${eventType}" event. Check the console logs for more details.`
                );
                ```

Reply:
ok you would catch the error inside the try/catch inside this try/catch?

Reply:
If I do it like that then the rest of the code in the sendEvent method does not have access to response

Reply:
I do want to point out that @erocheleau is putting it lightly with "I would". **You should** not have put the try-catch around the whole code, including the existing error handling. Referring to my previous comment, you effectively disabled most error handling (Promise rejects with error) and clobbered it into "Promise resolves with `undefined`", which is a completely new state integrators will have to deal with. This can't be the intent.

As a style remark, the fact that this is now relying on an implicit `return undefined` since there is no final return statement is bad form in my opinion. To me that's an indication that we're not dealing with the error, but just shoving it downstream via the "success" (resolve) path.